packages/x-date-pickers-pro/src/dateRangeViewRenderers/dateRangeViewRenderers.tsx

```diff
@@ -1,9 +1,12 @@
 import * as React from 'react';
 import { DateOrTimeView } from '@mui/x-date-pickers/models';
+import { DateOrTimeViewWithMeridiem } from '@mui/x-date-pickers/internals/models';
 import { DateRangeCalendar, DateRangeCalendarProps } from '../DateRangeCalendar';
 
-export interface DateRangeViewRendererProps<TDate, TView extends DateOrTimeView>
-  extends DateRangeCalendarProps<TDate> {
+export interface DateRangeViewRendererProps<
+  TDate,
+  TView extends DateOrTimeViewWithMeridiem | DateOrTimeView,
```

Comment:
Updated this generic rule everywhere for consistency, but essentially it allows to provide either views with or without the technically hidden `meridiem` view.

Reply:
Is there a downside just doing `TView extends DateOrTimeViewWithMeridiem` ? Since `DateOrTimeViewWithMeridiem` itself is a superset of `DateOrTimeView`.

Reply:
Great observation. I might have misjudged the initial implementation, but yeah, having only the widest set seems to do the trick nicely. 👌 🙏